### PR TITLE
Schema revisions

### PR DESCRIPTION
**Pull request by james-monkeyshines**
https://www.ebi.ac.uk/panda/jira/browse/ENSPROD-6332 goes into detail about the rationale behind these changes.

### CODE REVIEW

File: ensembl_metadata/models/genome.py [hunk 1]
```diff
@@ -70,41 +77,44 @@ class DatabaseType(models.TextChoices):
...
     name = models.CharField(max_length=255, unique=True)
 
     class Meta:
         db_table = 'dataset_source'
 
 
-class Dataset(models.Model):
-    class DatasetType(models.TextChoices):
+class DatasetType(models.Model):
+    class DatasetTopic(models.TextChoices):
```
Comment (marcoooo) [inline ensembl_metadata/models/genome.py:88]: Topic is good naming choice!

File: ensembl_metadata/models/genome.py [hunk 2]
```diff
@@ -59,7 +66,7 @@ class Meta:
 
 
 class DatasetSource(models.Model):
-    class DatabaseType(models.TextChoices):
+    class SourceType(models.TextChoices):
```
Comment (marcoooo) [inline ensembl_metadata/models/genome.py:69]: Source Type is much more accurate if this is not an actual DB

File: ensembl_metadata/models/genome.py [hunk 3]
```diff
@@ -1,160 +1,160 @@
...
+
+class DatasetType(models.Model):
+    class DatasetTopic(models.TextChoices):
+        ASSEMBLY = 'assembly'
+        GENESET = 'geneset'
+        ASSEMBLY_ANNOTATION = 'assembly_annotation'
+        GENESET_ANNOTATION = 'geneset_annotation'
+        COMPARATIVE = 'comparative'
+        REGULATION = 'regulation'
+        VARIATION = 'variation'
```
Comment (marcoooo) [inline ensembl_metadata/models/genome.py:95]: From CDM meeting, I think we should add a topic for XRef, unless your plan was to associate such things as "RefSeq" links to geneset_annotation. In such case, I don't know where we cold store this "reationship"

Comment (james-monkeyshines) [reply ensembl_metadata/models/genome.py:95]: I'd intended these as quite high-level groupings; cross-references would indeed be `geneset_annotation`s. My thinking was that we would have maybe two or three different xref dataset types (e.g. 'checksum_based_xref', 'alignment_based_xref'), with GO terms and protein domains also coming under this topic. I wasn't thinking that a dataset would be at the level of a xref source, e.g. 'refseq_xref', 'uniparc_xref' - that seems too granular. What we could/should do is have dataset_statistics with metadata that reflects the xref annotation, to reflect the provenance of the various source dbs, e.g. 'RefSeq count', 'RefSeq version'.

**Comment by james-monkeyshines** [comment]
The new schema is now being managed in another repo (https://github.com/Ensembl/ensembl-metadata-admin), with the grpc code elsewhere (https://github.com/Ensembl/ensembl-metadata-api).

**Pull request closed without merging**